phpBB/phpbb/filesystem/filesystem_interface.php

```diff
@@ -0,0 +1,284 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\filesystem;
+
+/**
+ * Interface for phpBB's filesystem service
+ */
+interface filesystem_interface
+{
+	/**
+	 * chmod all permissions flag
+	 *
+	 * @var int
+	 */
+	const CHMOD_ALL = 7;
+
+	/**
+	 * chmod read permissions flag
+	 *
+	 * @var int
+	 */
+	const CHMOD_READ = 4;
+
+	/**
+	 * chmod write permissions flag
+	 *
+	 * @var int
+	 */
+	const CHMOD_WRITE = 2;
+
+	/**
+	 * chmod execute permissions flag
+	 *
+	 * @var int
+	 */
+	const CHMOD_EXECUTE = 1;
+
+	/**
+	 * Change owner group of files/directories
+	 *
+	 * @param string|array|\Traversable	$files		The file(s)/directorie(s) to change group
+	 * @param string					$group		The group that should own the files/directories
+	 * @param bool 						$recursive	If the group should be changed recursively
+	 * @throws \phpbb\filesystem\exception\filesystem_exception	the filename which triggered the error can be
+	 * 															retrieved by filesystem_exception::get_filename()
+	 */
+	public function chgrp($files, $group, $recursive = false);
+
+	/**
+	 * Global function for chmodding directories and files for internal use
+	 *
+	 * The function accepts filesystem_interface::CHMOD_ flags in the permission argument
+	 * or the user can specify octal values (or any integer if it makes sense). All directories will have
+	 * an execution bit appended, if the user group (owner, group or other) has the any bit specified.
```

Comment:
* an execution bit appended, if the user group (owner, group or other) has any bit specified.